### ISSUE TITLE
Errors while restoring to a PwnageTool or Sn0wBreeze IPSW

### ISSUE THREAD

**Issue by TKO-Cuber** [axi0mX/ipwndfu#24]
iPhone 3GS New BR.

Any PwnageTool IPSW or Sn0wBreeze IPSW I try to restore to gives me error 1600/1601. The only Custom IPSWs that work are RedSn0w ones. I am using Sn0wBreeze for the capability to restore to a pre-jailbroken IPSW and use bootlogos.

I use `ipwndfu -p` to pwn dfu, then install the exploit using `ipwndfu -x` and re-pwn dfu. After that, I try to restore to an iOS 6.1.3 Sn0wBreeze IPSW, but I get error 1600/1601 every time. I am using iTunes version 11.0.5.5. The only custom IPSWs that work are the RedSn0w ones, but they are not pre-jailbroken so I'd have to jailbreak tethered using redsn0w and use p0sixspwn to untether it. I find Sn0wBreeze  to be better because the IPSWs are pre-jailbroken and have capability to use BootLogos.

What causes RedSn0w to work but PwnageTool and Sn0wbreeze not? Is there a fix? Thanks!

**Comment by axi0mX** [comment]
This is done in the wrong order.

This is the correct order:
1. ./ipwndfu -p
2. Restore IPSW in iTunes. Wait for it to finish.
3. ./ipwndfu -p
4. ./ipwndfu -x

**Comment by TKO-Cuber** [comment]
Ok sorry about that. A guy on Reddit said you have to install the exploit first... He's clearly wrong then.

I'll try it again and get back to you.

**Comment by TKO-Cuber** [comment]
Now I'm getting error 21. I'm trying to restore to 5.1.1, 6.1.3, or 4.3.3 and they all give me error 21.

**Comment by axi0mX** [comment]
Is your device in pwned DFU Mode?

**Comment by TKO-Cuber** [comment]
Yeah. I'll try again,

**Comment by axi0mX** [comment]
Did you create a 24Kpwn IPSW (i.e. for old bootrom 3GS)? When using redsn0w you must choose:

```
Q: Will this custom IPSW be used on a newer (fixed) version of the iPhone3GS?
A: No
```

**Comment by TKO-Cuber** [comment]
Yep. I chose no. Also I am using sn0wbreeze, is that ok? 

I'll try again with the restore.

**Comment by TKO-Cuber** [comment]
Ok. Tried Sn0wbreeze Old Br 6.1.3 IPSW and after it said preparing phone for restore it sent my 3GS into recovery mode and failed with error 21. That would be why I'm getting error 21, as error 21 occurs when you try to restore to an IPSW like this in recovery mode, but I don't know how I can fix this. 

Latest version of iTunes.

**Comment by axi0mX** [comment]
Try a redsn0w 24Kpwn IPSW, 4.3.3, 5.0, 5.0.1, 5.1, or 5.1.1.

**Comment by TKO-Cuber** [comment]
Any RedSn0w IPSW works. No sn0wbreeze works, and a Pwnagetool IPSW I tried for iOS 3.1.3 didn't work either.

**Comment by axi0mX** [comment]
Are you getting Error 28 with iOS 3.1.3?

Instead of ipwndfu, try using iREB for Mac (https://ih8sn0w.com/) to enter pwned DFU Mode before restoring a sn0wbreeze IPSW.

**Comment by TKO-Cuber** [comment]
Don't have a Mac...

**Comment by TKO-Cuber** [comment]
Is it possible to make a bootable USB with Mac on it to use iReb? Or can I download iReb using windows?

**Comment by axi0mX** [comment]
Use it on Windows. It's included in the menu in sn0wbreeze.

Are you getting Error 28 with iOS 3.1.3?

**Comment by TKO-Cuber** [comment]
I do. I know that it means the hardware isn't compatible. Good news is that I saw the restore screen with the loading bar on the phone, meaning the issue is just Sn0wBreeze. I'm putting iReb on my other PC on windows XP now,  as my current PC is on windows 10 and doesn't have the right .NET framework.

**Comment by TKO-Cuber** [comment]
I'll try the Sn0wbreeze 6.1.3 old BR IPSW with ireb used to PWN DFU now,

**Comment by axi0mX** [comment]
What week and year was your device made in? You can find based on digits 3,4,5 of the serial number.

**Comment by TKO-Cuber** [comment]
Digits are 103

**Comment by axi0mX** [comment]
Interesting. Week 3, 2011. This is the earliest made iPhone 3GS I heard of which is incompatible with iOS 3.

https://github.com/axi0mX/ipwndfu/blob/master/JAILBREAK-GUIDE.md#compatibility-with-older-ios-versions

**Comment by TKO-Cuber** [comment]
Just used iReb on my old PC, not a VM, and it said it worked. I tried to restore to a Sn0wBreeze Old BR 6.1.3 IPSW, but it still fails with error 21 and sends my device into recovery mode. This is pretty wierd. What are my options?

**Comment by axi0mX** [comment]
Use sn0wbreeze to make a custom IPSW for older iOS versions and try those.

**Comment by TKO-Cuber** [comment]
Ok

**Comment by TKO-Cuber** [comment]
Tried twice again using iReb, this time restoring to iOS 4.3.3 old BR Sn0wBreeze IPSW. Both times the phone showed the Sn0wBreeze logo, and after 5 seconds or so the regular apple logo came up, stayed for a second, and put me back in recovery mode with an error 20 from iTunes. I'll try again with other Firmwares... this is getting kinda frustrating...

**Comment by TKO-Cuber** [comment]
iOS 4.0- No Sn0wBreeze Logo- straight to recovery mode and error 20.
iOS 4.3.3- **Saw Sn0wBreeze logo** for a few seconds then into recovery mode with error 20.
iOS 5.1.1- No Sn0wBreeze Logo- straight to recovery mode and eror 21.
iOS 6.0-  Got past Sn0wBreeze Logo, even saw loading bar going. Stopped and gave error 14.
iOS 6.1.3- No Sn0wBreeze Logo- straight to recovery mode and error 21.

**Comment by TKO-Cuber** [comment]
In my opinion, 6.0 is the most promising since I got it to the bar. I'll mess around with 6.0 and see if I can get that to work.

**Comment by TKO-Cuber** [comment]
Still no luck... now the logo doesn't even come up for 6.0

**Comment by axi0mX** [comment]
Are you creating IPSWs with a custom boot logo? Try simply creating a jailbroken IPSW with default sn0wbreeze boot logos and see if that works.

**Comment by TKO-Cuber** [comment]
So I tried that and also increased the root partition to 2013 MB and that made it work. Now I'm going to try to restore to a NOR-only IPSW with bootlogos and see what happens.

**Comment by TKO-Cuber** [comment]
I remember reading that ipwndfu will eventually allow for boot logo changes and verbose boot... is this still happening?

**Comment by axi0mX** [comment]
I might add that feature in the future, but there is no ETA.

Does a NOR-only IPSW with custom bootlogos from sn0wbreeze work for you?

**Comment by TKO-Cuber** [comment]
No, but I used the old Sn0wBreeze for iOS 3, so I'd figure it would make an iOS 3 NOR-only IPSW, which my device is not compatible with. Don't remember if I got error 28 or something else though. I'll try again hen I get home.

What other tools can I use to make a NOR-only IPSW?

**Comment by axi0mX** [comment]
I don't know.

**Comment by iPhoneDowngrader** [comment]
If you have not figured it out, try using a macOS virtual machine to make the IPSW, I made one for iOS 3 and its currently on my google drive because thats how I got it out of the VM. Link: https://drive.google.com/file/d/0Bx2X3o1VIqhEa2FRcjVvQmdwZUU/view?usp=sharing

**Comment by axi0mX** [comment]
Please: no sharing of IPSW links. It is a bad idea for security reasons (It is trivial to add malware to an IPSW) and it is not legal. Only get IPSWs from Apple.
